x509util/extensions.go

```diff
@@ -0,0 +1,473 @@
+package x509util
+
+import (
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+func convertName(s string) string {
```

Comment:
I'm confused why we need this function in this file when none of the strings you're using it on have any "_".

Reply:
I did this because in the initial spec strings like `DigitalSignature` were written as `digital_signature`. To support both cases. It depends on how strictly do we want to be.